x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -1356,39 +1415,108 @@ public static void registerSettings(List<Setting<?>> settings) {
          * a singleton list of the empty string ([""]).
          */
         EventFilterPolicy(String name, Predicate<String> ignorePrincipalsPredicate, Predicate<String> ignoreRealmsPredicate,
-                Predicate<String> ignoreRolesPredicate, Predicate<String> ignoreIndicesPredicate) {
+                Predicate<String> ignoreRolesPredicate, Predicate<String> ignoreIndicesPredicate,
+                Predicate<String> ignoreIndexPrivilegesPredicate, Predicate<String> ignoreClusterPrivilegesPredicate) {
             this.name = name;
             // "null" values are "unexpected" and should not match any ignore policy
             this.ignorePrincipalsPredicate = ignorePrincipalsPredicate;
             this.ignoreRealmsPredicate = ignoreRealmsPredicate;
             this.ignoreRolesPredicate = ignoreRolesPredicate;
             this.ignoreIndicesPredicate = ignoreIndicesPredicate;
+            this.ignoreIndexPrivilegesPredicate = ignoreIndexPrivilegesPredicate;
+            this.ignoreClusterPrivilegesPredicate = ignoreClusterPrivilegesPredicate;
+
+            if (ignoreIndexPrivilegesPredicate.test("") == false && ignoreClusterPrivilegesPredicate.test("") == false) {
+                final String message = String.format(
+                    Locale.ROOT,
+                    "Both Index and Cluster privilege ignore filters are set for policy [%s]. " +
+                        "Please update your configuration settings and remove one of these filters to avoid ambiguity.", this.name);
+                throw new IllegalArgumentException(message);
+            }
+
+            if (ignoreClusterPrivilegesPredicate.test("") == false) {
+                ClusterPermission.Builder builder = ClusterPermission.builder();
+                String[] clusterPrivileges = ignoreClusterPrivilegesPredicate.toString().split("\\|");
```

Comment:
I think we should not rely on the output of `toString` for logic.

What's wrong with iterating the list of values, and then try to build the cluster permission if the value is not a wildcard or a regex? If it is a wildcard or a regex we add them to the automaton.

There's also a question here whether wildcards and regex are interpreted as matching actions and/or privileges.
I would say that wildcards (and regexps) only work for actions, i.e. `m*` is not expanded to cover all privileges that start with "m" but all actions that start with "m" (no action). I'm OK with the other option too, but we should document these.